src/strategyHelpers/blueGreen/blueGreenHelper.test.ts

```diff
@@ -0,0 +1,128 @@
+import {
+   createWorkloadsWithLabel,
+   deleteWorkloadsAndServicesWithLabel,
+   getManifestObjects,
+   getNewBlueGreenObject,
+   GREEN_LABEL_VALUE,
+   isServiceRouted,
+   NONE_LABEL_VALUE
+} from './blueGreenHelper'
+import * as bgHelper from './blueGreenHelper'
+import {Kubectl} from '../../types/kubectl'
+import * as fileHelper from '../../utilities/fileUtils'
+
+jest.mock('../../types/kubectl')
+
+describe('bluegreenhelper functions', () => {
+   let testObjects
+   beforeEach(() => {
+      //@ts-ignore
+      Kubectl.mockClear()
+      testObjects = getManifestObjects(['test/unit/manifests/test-ingress.yml'])
+
+      jest
+         .spyOn(fileHelper, 'writeObjectsToFile')
+         .mockImplementationOnce(() => [''])
+   })
+
+   test('it should parse objects correctly from one file', () => {
+      expect(testObjects.deploymentEntityList[0].kind).toBe('Deployment')
+      expect(testObjects.serviceEntityList[0].kind).toBe('Service')
+      expect(testObjects.ingressEntityList[0].kind).toBe('Ingress')
+
+      expect(
+         testObjects.deploymentEntityList[0].spec.selector.matchLabels.app
+      ).toBe('nginx')
+   })
+
+   test('correctly makes new blue green object', () => {
+      const modifiedDeployment = getNewBlueGreenObject(
+         testObjects.deploymentEntityList[0],
+         GREEN_LABEL_VALUE
+      )
+      //@ts-ignore
```

Comment:
what's the reason for having the ts-ignores in the expect? that likely should be fixed instead of ignored

Reply:
I'm working on fixing return types in the refactor if that's ok, just because there's kind of a domino effect with changing functions since their usage is pretty scattered